logical/plugin/backend_client.go

```diff
@@ -0,0 +1,204 @@
+type HandleExistenceCheckArgs struct {
+	StorageID uint32
+	Request   *logical.Request
+}
+
+// HandleExistenceCheckReply is the reply for HandleExistenceCheck method.
+type HandleExistenceCheckReply struct {
+	CheckFound bool
+	Exists     bool
+	Error      *plugin.BasicError
+}
+
+// SetupArgs is the args for Setup method.
+type SetupArgs struct {
+	StorageID uint32
+	LoggerID  uint32
+	SysViewID uint32
+	Config    map[string]string
+}
+
+// SetupReply is the reply for Setup method.
+type SetupReply struct {
+	Error *plugin.BasicError
+}
+
+// TypeReply is the reply for the Type method.
+type TypeReply struct {
+	Type logical.BackendType
+}
+
+func (b *backendPluginClient) HandleRequest(req *logical.Request) (*logical.Response, error) {
+	args := &HandleRequestArgs{
+		Request: req,
+	}
+	var reply HandleRequestReply
+
+	err := b.client.Call("Plugin.HandleRequest", args, &reply)
+	if err != nil {
+		return nil, err
+	}
+	if reply.Error != nil {
+		if reply.Error.Error() == logical.ErrUnsupportedOperation.Error() {
+			return nil, logical.ErrUnsupportedOperation
+		}
+		return nil, reply.Error
+	}
+
+	return reply.Response, nil
+}
+
+func (b *backendPluginClient) SpecialPaths() *logical.Paths {
+	var reply SpecialPathsReply
+	// var paths logical.Paths
+	err := b.client.Call("Plugin.SpecialPaths", new(interface{}), &reply)
+	if err != nil {
+		return nil
+	}
+
+	return reply.Paths
+}
+
+// System returns vault's system view. The backend client stores the view during
+// Setup, so there is no need to shim the system just to get it back.
+func (b *backendPluginClient) System() logical.SystemView {
+	return b.system
+}
+
+// Logger returns vault's logger. The backend client stores the logger during
+// Setup, so there is no need to shim the logger just to get it back.
+func (b *backendPluginClient) Logger() log.Logger {
+	return b.logger
+}
+
+func (b *backendPluginClient) HandleExistenceCheck(req *logical.Request) (bool, bool, error) {
+	args := &HandleExistenceCheckArgs{
+		Request: req,
+	}
+	var reply HandleExistenceCheckReply
+
+	err := b.client.Call("Plugin.HandleExistenceCheck", args, &reply)
+	if err != nil {
+		return false, false, err
+	}
+	if reply.Error != nil {
+		// THINKING: Should be be a switch on all error types?
+		if reply.Error.Error() == logical.ErrUnsupportedPath.Error() {
+			return false, false, logical.ErrUnsupportedPath
+		}
+		return false, false, reply.Error
+	}
+
+	return reply.CheckFound, reply.Exists, nil
+}
+
+func (b *backendPluginClient) Cleanup() {
+	b.client.Call("Plugin.Cleanup", new(interface{}), &struct{}{})
+}
+
+func (b *backendPluginClient) Initialize() error {
+	err := b.client.Call("Plugin.Initialize", new(interface{}), &struct{}{})
+	return err
+}
+
+func (b *backendPluginClient) InvalidateKey(key string) {
+	b.client.Call("Plugin.InvalidateKey", key, &struct{}{})
+}
+
+func (b *backendPluginClient) Setup(config *logical.BackendConfig) error {
```

Comment:
Could you add a comment to this function stating that the backend is actually instantiated here?